constraint/pkg/apis/templates/v1beta1/conversion.go

```diff
@@ -24,8 +24,14 @@ import (
 )
 
 func Convert_v1beta1_Validation_To_templates_Validation(in *Validation, out *coreTemplates.Validation, s conversion.Scope) error { //nolint:golint
-	if in.OpenAPIV3Schema != nil {
-		inSchemaCopy := in.OpenAPIV3Schema.DeepCopy()
+	inSchema := in.OpenAPIV3Schema
+	// legacySchema should allow for users to provide arbitrary parameters, regardless of whether the user specified them
+	if in.LegacySchema && inSchema == nil {
+		inSchema = &apiextensionsv1.JSONSchemaProps{}
```

Comment:
This strikes me as a defaulting action, rather than a conversion action.  I think we'd be better off linking those two things together somehow.

Reply:
This is a conversion action.

In the case where `legacySchema == false`, the schema remains nil and the expectation is that no parameters are provided.

When `legacySchema == true`, users could provide any value to parameters and it would be accepted, this preserves that.

Essentially we are unpacking the implicit `x-preserves` embedded in `legacySchema`, no new intent is being injected (which defaulting would do)